src/OpalCompiler-Core/IRBytecodeGenerator.class.st

```diff
@@ -203,8 +209,10 @@ IRBytecodeGenerator >> compiledMethodWith: trailer [
 
 { #category : #results }
 IRBytecodeGenerator >> compiledMethodWith: trailer header: header literals: lits [
-	| cm |
-	cm := trailer createMethod: self bytecodes size class: CompiledMethod header: header.
+	| cm cmClass|
+	"we look up the class in the productionEnvironment"
+	cmClass := compilationContext productionEnvironment at: #CompiledMethod ifAbsent: [ CompiledMethod ].
```

Comment:
Like here. Why "compilationContext environment at: #CompiledMethod" is not enough?

Reply:
I think I see. 
For example in TelePharo the remote playground doIt's are compiled locally using remote environment for globals. But method is created as local object using local CompiledMethod class. 
At the end method is sent to remote side for execution.

In other scenario we would want to use different method class. For example we could copy CompiledMethod and modify its logic for experiment. Then we will need to use it to test compilation

Reply:
yes, exactly.

Reply:
Now I wonder what other objects you are going to retrieve from this productionEnvironment?
Or in other words why simple variable #compilerMethodClass is not enough?

Reply:
The idea is that this might be extended to all the Literal objects that are part of the compiledMethod that is created: Symbol, String, Array, ....